R/waapwlspetpeese.R

```diff
@@ -0,0 +1,610 @@
+        ))
+    } else if (type == "petPeese") {
+      # PET
+      effectTest$addRows(list(
+        type = "PET",
+        stat = models[["pet"]]["(Intercept)", "t value"],
+        df   = attr(models[["pet"]], "df"),
+        pVal = models[["pet"]]["(Intercept)", "Pr(>|t|)"]
+      ))
+    }
+
+  }
+
+
+  ### test of bias
+  if (type == "petPeese") {
+    biasTest <- createJaspTable(title = gettext("Test of Publication Bias"))
+    biasTest$position <- 2
+    fitTests[["biasTest"]] <- biasTest
+
+    biasTest$addColumnInfo(name = "type",  title = "",                type = "string")
+    biasTest$addColumnInfo(name = "stat",  title = gettext("t"),      type = "number")
+    biasTest$addColumnInfo(name = "df",    title = gettext("df"),     type = "integer")
+    biasTest$addColumnInfo(name = "pVal",  title = gettext("p"),      type = "pvalue")
+
+    if (!is.null(models)) {
+
+      biasTest$addRows(list(
+        type = "PET",
+        stat = models[["pet"]]["PET", "t value"],
+        df   = attr(models[["pet"]], "df"),
+        pVal = models[["pet"]]["PET", "Pr(>|t|)"]
+      ))
+
+    }
+  }
+
+  return()
+}
+.wwppMakeEstimatesTables     <- function(jaspResults, dataset, options, type) {
+
+  models   <- jaspResults[["models"]]$object
+
+  ### assuming heterogeneity
+  if (is.null(jaspResults[["estimates"]])) {
+    # create container
+    estimates <- createJaspContainer(title = gettext("Estimates"))
+    estimates$position <- 2
+    estimates$dependOn(c(.wwppDependencies, "estimates"))
+    jaspResults[["estimates"]] <- estimates
+  } else {
+    estimates <- jaspResults[["estimates"]]
+  }
+
+  # mean estimates
+  if (is.null(estimates[["mean"]]) && options[["estimatesMean"]]) {
+    estimatesMean <- createJaspTable(title = gettextf(
+      "Mean Estimates(%s)",
+      if (options[["measures"]] == "correlation") "\u03C1" else "\u03BC"
+    ))
+    estimatesMean$position <- 1
+    estimates$dependOn(c("estimatesMean"))
+    estimates[["estimatesMean"]] <- estimatesMean
+    estimatesMean <- .wwppFillEstimates(jaspResults, estimatesMean, models, options, type)
+  }
+
+  # PET-PEESE estimates
+  if (type == "petPeese") {
+    if (is.null(estimates[["petPeese"]]) && options[["estimatesPetPeese"]]) {
+      petPeese <- createJaspTable(title = gettext("PET-PEESE Regression Estimates"))
+      petPeese$position  <- 2
+      petPeese$dependOn(c("estimatesPetPeese"))
+      estimates[["petPeese"]] <- petPeese
+      petPeese <- .wwppFillPetPeese(jaspResults, petPeese, models, options)
+    }
+  }
+
+  # heterogeneity estimates
+  if (is.null(estimates[["heterogeneity"]]) && options[["estimatesSigma"]]) {
+    heterogeneity <- createJaspTable(title = gettext("Multiplicative Heterogeneity Estimates"))
+    heterogeneity$position <- 3
+    heterogeneity$dependOn(c("estimatesSigma"))
+    estimates[["heterogeneity"]] <- heterogeneity
+    heterogeneity <- .wwppFillHeterogeneity(jaspResults, heterogeneity, models, options, type)
+  }
+
+  return()
```

Comment:
No `return()` required.

Reply:
yes, but I would like to keep it here as well :)